web/src/containers/Main/hooks/index.ts

```diff
@@ -55,20 +64,63 @@ export function useMain(questionnaireId: string) {
         [questionnaireId, fhirMode],
     );
 
+    const showModal = (response: any, type: string) => {
+        if (type === 'success') {
+            toast.success('New mapper created');
+        } else {
+            toast.error(
+                formatError(response.error, {
+                    mapping: { conflict: 'Please reload page' },
+                    format: (errorCode, errorDescription) =>
+                        `An error occurred: ${errorDescription} (${errorCode}). Please reach tech support`,
+                }),
+            );
+        }
+    };
+
     const saveQuestionnaireFHIR = useCallback(
         async (resource: Questionnaire) => {
-            const response = await service({
+            const response = await service<unknown, OperationOutcome>({
                 method: 'PUT',
                 data: resource,
                 url: `/${fhirMode ? 'fhir/' : ''}Questionnaire/${resource.id}`,
             });
+
             if (isSuccess(response)) {
                 questionnaireManager.reload();
             } else {
-                console.error('Could not save Questionnaire:', response.error.toString());
+                if (
+                    response.error.issue[0] &&
+                    response.error.issue[0].expression &&
+                    response.error.issue[0].expression[0].slice(0, 21) === 'Questionnaire.mapping' &&
+                    response.error.issue[0].code === 'invalid' &&
+                    response.error.resourceType === 'OperationOutcome'
+                ) {
+                    const mappingId =
+                        response.error.issue[0].diagnostics &&
+                        response.error.issue[0].diagnostics.split(' ')[2].slice(8);
```

Comment:
Diagnostic is a free text message, you can't rely on this text.
Could you please use `issue.expression` to extract the index of mapping and use it to get the `id` from `Questionnaire.mapping` array?